api/src/main/java/io/opentelemetry/trace/SpanContext.java

```diff
@@ -75,25 +76,88 @@ public static SpanContext create(
    * @since 0.1.0
    */
   public static SpanContext createFromRemoteParent(
-      TraceId traceId, SpanId spanId, TraceFlags traceFlags, TraceState traceState) {
-    return new AutoValue_SpanContext(traceId, spanId, traceFlags, traceState, /* remote=*/ true);
+      CharSequence traceId, CharSequence spanId, TraceFlags traceFlags, TraceState traceState) {
+    return create(traceId, 0, spanId, 0, traceFlags, traceState, /* remote=*/ true);
   }
 
+  /**
+   * Creates a new {@code SpanContext} with the given identifiers and options.
+   *
+   * @param traceId a CharSequence containing the trace identifier of the span context.
+   * @param traceIdOffset the offset at which the traceId starts.
+   * @param spanId the CharSequence containing the the span identifier of the span context.
+   * @param spanIdOffset the offset at which the spanId starts.
+   * @param traceFlags the trace options for the span context.
+   * @param traceState the trace state for the span context.
+   * @return a new {@code SpanContext} with the given identifiers and options.
+   * @since 0.1.0
+   */
+  @SuppressWarnings({
+    "InconsistentOverloads",
+    "UngroupedOverloads",
+    "OverloadMethodsDeclarationOrder"
+  })
+  public static SpanContext create(
+      CharSequence traceId,
+      int traceIdOffset,
+      CharSequence spanId,
+      int spanIdOffset,
+      TraceFlags traceFlags,
+      TraceState traceState,
+      boolean remote) {
+    return new AutoValue_SpanContext(
+        traceId.subSequence(traceIdOffset, traceIdOffset + 32).toString(),
```

Comment:
I don't know if the offset versions provide enough user benefit to have (not much harder to call subsequence than grok this long method signature). There doesn't seem to be a performance benefit since we always need to take the substring anyways.

If we only had two factories, than I think we could also provide two symmetric factories that accept `byte[]` and we don't need public APIs like `TraceId.toLowerBase16(byte[])`

Reply:
Though realized a byte[] version of this factory would have benefit from an offset version :)

Reply:
This was a method signature that @bogdandrutu had requested. I don't know that it's necessary at the moment, though.

Reply:
I think we should get rid of the offsets for now; we can always add another option later; we can't take it back if we leave it in.

Reply:
I got rid of them. Smaller API is better for now.